FINE_filtering.py

```diff
@@ -21,7 +21,7 @@ def fine_filtering(templates, reads, k, f):
         kmer_matches = [0]*number_of_kmers
 
         for read in reads:
-            kmer_reads = [read[i:i + k] for i in range(0, length_of_read - k + 1)]
+            kmer_reads = set([read[i:i + k] for i in range(0, length_of_read - k + 1)])
```

Comment:
The set should not be taken here, since information could be lost. For example, if the template is aabb, and the read is bbaabbaa, only half the point will be counted.

Reply:
I'm confused by your example.

Reply:
so, our read is bbaaabbaa, let's say k=3, so, the set of k-mers is bba, baa, aab, abb, bba, baa.
So, in this toy example the k-mers bba and baa occur twice, and represent two different template units. If you take a set of k-mers, you will have  bba, baa, aab, abb, and so will be undercounting.

Reply:
We're counting the number of template k-mers that are found in the read - not the number of read k-mers that are found in the template though.